common/guestchain/src/manager.rs

```diff
@@ -264,6 +278,18 @@ impl<PK: crate::PubKey> ChainManager<PK> {
         })
     }
 
+    fn validate_generate_next_epoch(
```

Comment:
`should_generate_next_epoch`?

Reply:
i removed the method and replaced it to take an immutable ref instead.